### Task 1 -  Create a panda series

In [5]:
price = [80121, 48483, 37077, 30660, 68536, 52666, 78703, 51036, 55643, 56990, 39967, 93822, 51406, 93947, 78373, 62765,
         47536, 40528, 58780, 83197]

In [6]:
import pandas as pd

In [7]:
index = pd.Series(range(101,121))
print(index)

0     101
1     102
2     103
3     104
4     105
5     106
6     107
7     108
8     109
9     110
10    111
11    112
12    113
13    114
14    115
15    116
16    117
17    118
18    119
19    120
dtype: int64


In [8]:
car_price = pd.Series(price,index)

In [9]:
car_price

101    80121
102    48483
103    37077
104    30660
105    68536
106    52666
107    78703
108    51036
109    55643
110    56990
111    39967
112    93822
113    51406
114    93947
115    78373
116    62765
117    47536
118    40528
119    58780
120    83197
dtype: int64

Interpretation - Here i have created a panda series and store the value as indexes and car prices. the index is from 101 to 120
and the prices of the cars are already given in the data

### Task 2 -  row indexes and highest and lowest prices of the cars

In [10]:
df = pd.DataFrame() 

In [11]:
df["Index"] = index.values
df["price"] = price

In [12]:
df

Index  price
0     101  80121
1     102  48483
2     103  37077
3     104  30660
4     105  68536
5     106  52666
6     107  78703
7     108  51036
8     109  55643
9     110  56990
10    111  39967
11    112  93822
12    113  51406
13    114  93947
14    115  78373
15    116  62765
16    117  47536
17    118  40528
18    119  58780
19    120  83197

In [13]:
df1 = df.sort_values(by='price')

In [14]:
df1

Index  price
3     104  30660
2     103  37077
10    111  39967
17    118  40528
16    117  47536
1     102  48483
7     108  51036
12    113  51406
5     106  52666
8     109  55643
9     110  56990
18    119  58780
15    116  62765
4     105  68536
14    115  78373
6     107  78703
0     101  80121
19    120  83197
11    112  93822
13    114  93947

In [15]:
df1.iloc[19]

Index      114
price    93947
Name: 13, dtype: int64

In [16]:
df1

Index  price
3     104  30660
2     103  37077
10    111  39967
17    118  40528
16    117  47536
1     102  48483
7     108  51036
12    113  51406
5     106  52666
8     109  55643
9     110  56990
18    119  58780
15    116  62765
4     105  68536
14    115  78373
6     107  78703
0     101  80121
19    120  83197
11    112  93822
13    114  93947

In [17]:
df1.iloc[0]

Index      104
price    30660
Name: 3, dtype: int64

Interpretation - Here i have displayed the row indexes and highest and lowest prices of the cars by using sort() and iloc functions

### Task 3 - Display the price of the top 5 cars with the highest value ,along with row index 

In [18]:
df2 = df.sort_values(by='price', ascending = False)
df2

Index  price
13    114  93947
11    112  93822
19    120  83197
0     101  80121
6     107  78703
14    115  78373
4     105  68536
15    116  62765
18    119  58780
9     110  56990
8     109  55643
5     106  52666
12    113  51406
7     108  51036
1     102  48483
16    117  47536
17    118  40528
10    111  39967
2     103  37077
3     104  30660

In [19]:
top_five = df2[0:5]

In [20]:
top_five

Index  price
13    114  93947
11    112  93822
19    120  83197
0     101  80121
6     107  78703

Interpretation - Here ,i have Displayed the price of the top 5 cars with the highest value ,along with row index

### Task 4 - Display the price of all cars price between 30 - 40k along with the row indexes 

In [21]:
price_bet = df[(df['price'] > 30000) & (df['price'] < 40000)]


In [22]:
print(price_bet)

    Index  price
2     103  37077
3     104  30660
10    111  39967


In [23]:
price_bet.shape

(3, 2)

Interpretation - Here i have displayed the price of all cars price between 30 - 40k along with the row indexes and the count of the cars is 3  

### Task 5 - Dsiplay the price and row index of those cars whose price are higher than the average price and display the count of those cars 

In [24]:
avg = df.price.mean()
avg

60511.8

In [25]:
high_avg =df[df["price"]>avg]

In [26]:
high_avg

Index  price
0     101  80121
4     105  68536
6     107  78703
11    112  93822
13    114  93947
14    115  78373
15    116  62765
19    120  83197

In [27]:
high_avg.count()

Index    8
price    8
dtype: int64

Interpretation - Here , i have identify the average price of all cars and then identify the cars price which are higher than the average price of cars and the count of those cars are 8

### Task 6 -  add new cars  

In [28]:
new_cars = pd.DataFrame({"Index":[201,202,203],"price":[34000,45000,54000]})

In [29]:
new_cars

Index  price
0    201  34000
1    202  45000
2    203  54000

In [30]:
all_cars = pd.concat([df, new_cars],ignore_index =True)

In [31]:
all_cars

Index  price
0     101  80121
1     102  48483
2     103  37077
3     104  30660
4     105  68536
5     106  52666
6     107  78703
7     108  51036
8     109  55643
9     110  56990
10    111  39967
11    112  93822
12    113  51406
13    114  93947
14    115  78373
15    116  62765
16    117  47536
17    118  40528
18    119  58780
19    120  83197
20    201  34000
21    202  45000
22    203  54000

Interpretation - Here , i have created a new table and concatenated the new table with existing table

### Task 7 - The car dealer is offering 20% discount for the cars whose price are above 60000 

In [32]:
pr = all_cars[all_cars["price"]>60000]

In [33]:
pr

Index  price
0     101  80121
4     105  68536
6     107  78703
11    112  93822
13    114  93947
14    115  78373
15    116  62765
19    120  83197

In [34]:
a = pr.price*0.8                                                           

In [73]:
a

0     64096.8
4     54828.8
6     62962.4
11    75057.6
13    75157.6
14    62698.4
15    50212.0
19    66557.6
Name: price, dtype: float64

In [89]:
Discounted_price = pd.DataFrame(a,index=None)

In [92]:
s = Discounted_price.rename(columns={"price" :"Discounted_price"})
s


Discounted_price
0            64096.8
4            54828.8
6            62962.4
11           75057.6
13           75157.6
14           62698.4
15           50212.0
19           66557.6

In [93]:
discount =  pd.concat([all_cars,s],axis=1)

In [94]:
discount

Index  price  Discounted_price
0     101  80121           64096.8
1     102  48483               NaN
2     103  37077               NaN
3     104  30660               NaN
4     105  68536           54828.8
5     106  52666               NaN
6     107  78703           62962.4
7     108  51036               NaN
8     109  55643               NaN
9     110  56990               NaN
10    111  39967               NaN
11    112  93822           75057.6
12    113  51406               NaN
13    114  93947           75157.6
14    115  78373           62698.4
15    116  62765           50212.0
16    117  47536               NaN
17    118  40528               NaN
18    119  58780               NaN
19    120  83197           66557.6
20    201  34000               NaN
21    202  45000               NaN
22    203  54000               NaN

In [95]:
discount

Index  price  Discounted_price
0     101  80121           64096.8
1     102  48483               NaN
2     103  37077               NaN
3     104  30660               NaN
4     105  68536           54828.8
5     106  52666               NaN
6     107  78703           62962.4
7     108  51036               NaN
8     109  55643               NaN
9     110  56990               NaN
10    111  39967               NaN
11    112  93822           75057.6
12    113  51406               NaN
13    114  93947           75157.6
14    115  78373           62698.4
15    116  62765           50212.0
16    117  47536               NaN
17    118  40528               NaN
18    119  58780               NaN
19    120  83197           66557.6
20    201  34000               NaN
21    202  45000               NaN
22    203  54000               NaN

Interpretation - Here i have displayed the cars prices after discount where car price grater than 60000 and the discount is 20%

### Task 8 

In [96]:
discount

Index  price  Discounted_price
0     101  80121           64096.8
1     102  48483               NaN
2     103  37077               NaN
3     104  30660               NaN
4     105  68536           54828.8
5     106  52666               NaN
6     107  78703           62962.4
7     108  51036               NaN
8     109  55643               NaN
9     110  56990               NaN
10    111  39967               NaN
11    112  93822           75057.6
12    113  51406               NaN
13    114  93947           75157.6
14    115  78373           62698.4
15    116  62765           50212.0
16    117  47536               NaN
17    118  40528               NaN
18    119  58780               NaN
19    120  83197           66557.6
20    201  34000               NaN
21    202  45000               NaN
22    203  54000               NaN

In [97]:
discount.count()

Index               23
price               23
Discounted_price     8
dtype: int64

In [98]:
sold = discount.drop([0,4,6,11,13,14,15,19])

In [99]:
sold

Index  price  Discounted_price
1     102  48483               NaN
2     103  37077               NaN
3     104  30660               NaN
5     106  52666               NaN
7     108  51036               NaN
8     109  55643               NaN
9     110  56990               NaN
10    111  39967               NaN
12    113  51406               NaN
16    117  47536               NaN
17    118  40528               NaN
18    119  58780               NaN
20    201  34000               NaN
21    202  45000               NaN
22    203  54000               NaN

In [100]:
sold.count()

Index               15
price               15
Discounted_price     0
dtype: int64

Interpretation- Above i have displayed that the discount cars has been sold and count of cars before sold is 23 and count of cars after sold is 15  .so total 8 cars has been sold after the discount placed for the cars